### PR TITLE
[SYCL] Test multi-target compilation for SYCL2020 spec constants

### PR DESCRIPTION
**Pull request by AGindinson**
Tests https://github.com/intel/llvm/pull/4591.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

**Comment by AGindinson** [comment]
CUDA reduction failures are the known unrelated ones, otherwise the tests behave [as expected](http://pre-commit.llvm.intel.com:8013/file_review?path_name=/llvm_test_suite_associate/PR4591/ad0044a37e), passing on Lin/Win with the product PR and failing without it.

File: SYCL/SpecConstants/2020/non_native/multiple-targets.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// REQUIRES: opencl-aot, cpu
+// UNSUPPORTED: cuda
+
+// RUN: %clangxx -fsycl -fsycl-targets=spir64,spir64_x86_64 %S/Inputs/common.cpp -o %t_spv_cpu.out
+// RUN: %CPU_RUN_PLACEHOLDER %t_spv_cpu.out
+// RUN: %GPU_RUN_PLACEHOLDER %t_spv_cpu.out
+// Ensure that image ordering does not impact the execution
```
Comment (kbobrovs) [inline SYCL/SpecConstants/2020/non_native/multiple-targets.cpp:7]: nit: I always find "image" kind of misleading to represent a "picture". `Binary image` or `device binary` would sound better for my subjective ear.

**Review by kbobrovs — APPROVED** [review]

File: SYCL/SpecConstants/2020/non_native/multiple-targets.cpp (hunk 1 above)
Comment (AGindinson) [reply SYCL/SpecConstants/2020/non_native/multiple-targets.cpp:7]: Fair point, I agree this naming is a bit confusing. There are a few more places in the test suite that use it, though, so I'll submit a separate PR to make sure we use the "clear" naming everywhere.

**Comment by AGindinson** [comment]
@intel/llvm-gatekeepers Could you please merge this one so that I could proceed with a [follow-up PR](https://github.com/intel/llvm-test-suite/pull/584#discussion_r759024645)?

**Pull request merged**